eng/scripts/Language-Settings.ps1

```diff
@@ -260,6 +260,260 @@ function Update-java-CIConfig($pkgs, $ciRepo, $locationInDocRepo, $monikerId=$nu
   Set-Content -Path $pkgJsonLoc -Value $jsonContent
 }
 
+$PackageExclusions = @{ 
+
+}
+
+# Validates if the package will succeed in the CI build by validating the 
+# existence of a com folder in the unzipped source package
+function SourcePackageHasComFolder($package, $workingDirectory) {
+  $packageDirectory = Join-Path `
+    $workingDirectory `
+    "$($package.packageGroupId)__$($package.packageArtifactId)__$($package.packageVersion)"
+  New-Item -ItemType Directory -Path $packageDirectory -Force
+
+  try
+  {
+    $packageArtifact = "$($package.packageGroupId):$($package.packageArtifactId):$($package.packageVersion):jar:sources"
+    $mvnResults = mvn `
+      dependency:copy `
+      -Dartifact="$packageArtifact" `
+      -DoutputDirectory="$packageDirectory"
+
+    if ($LASTEXITCODE) { 
+      LogWarning "Could not download source artifact: $packageArtifact"
+      # TODO: Better outputting
+      $mvnResults | Write-Host
+      return $false
+    }
+
+    $sourcesJarPath = (Get-ChildItem -File -Path $packageDirectory -Filter "*-sources.jar")[0]
+    $sourcesExtractPath = Join-Path $packageDirectory "sources"
+    Add-Type -AssemblyName System.IO.Compression.FileSystem
+    [System.IO.Compression.ZipFile]::ExtractToDirectory($sourcesJarPath, $sourcesExtractPath)
+
+    if (!(Test-Path "$sourcesExtractPath\com")) {
+      LogWarning "Could not locate 'com' folder extracting $packageArtifact"
+      return $false
+    }
+  }
+  catch 
+  {
+    LogError "Exception while updating checking if package can be documented: $($package.packageGroupId):$($package.packageArtifactId)"
+    LogError $_ 
+    LogError $_.ScriptStackTrace
+    return $false
+  }
+
+  return $true
+}
+
+function PackageDependenciesResolve($package, $workingDirectory) { 
+  $packageDirectory = Join-Path `
+    $workingDirectory `
+    "$($package.packageGroupId)__$($package.packageArtifactId)__$($package.packageVersion)"
+  New-Item -ItemType Directory -Path $packageDirectory -Force
+  
+  $pomArtifactName = "$($package.packageGroupId):$($package.packageArtifactId):$($package.packageVersion):pom"
+  $artifactDownloadOutput = mvn `
```

Comment:
Is this duplicated between here and SourcePackageHasComFolder? Should we only be running it once or is there a difference I cannot easily see?

Reply:
The difference is at the end: 

Download the sources jar (`...:jar:sources`) -- 
```
"$($package.packageGroupId):$($package.packageArtifactId):$($package.packageVersion):jar:sources"
```

Download the pom (`...:pom`)  --
```
"$($package.packageGroupId):$($package.packageArtifactId):$($package.packageVersion):pom"
```

However the leading portion could be created in a higher level function as well as the package-specific temp directory.